### PR TITLE
wip 0.3.0

### PR DESCRIPTION
**Pull request by chrisguida**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,6 +27,8 @@ ADD ./configurator.py /configurator.py
 RUN chmod a+x /configurator.py
 ADD ./renew.sh /usr/local/bin/renew.sh
 RUN chmod a+x /usr/local/bin/renew.sh
+ADD ./config.sh /usr/local/bin/config
```
Comment (elvece) [inline Dockerfile:30]: no longer needed as can just use the `config` command in the`compat` image

File: config.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -e
```
Comment (elvece) [inline config.sh:1]: this file can be deleted

File: manifest.yaml [hunk 3]
```diff
@@ -1,38 +1,99 @@
...
-release-notes: |
-  - resolves issues where one busy room could cause all federated traffic to fall behind
-  - Various other features and bugfixes: https://github.com/matrix-org/synapse/releases/tag/v1.37.1
+assets:
+  license: LICENSE
+  icon: icon.png
+  instructions: instructions.md
+  docker-images: image.tar
+  icon-type: png
+main:
```
Comment (elvece) [inline manifest.yaml:22]: add key:  `name` `description` to each interface (just "main" in this case)

File: Makefile [hunk 4]
```diff
@@ -8,16 +8,16 @@ all: synapse.s9pk
 install: synapse.s9pk
 	appmgr install synapse.s9pk
 
-synapse.s9pk: manifest.yaml config_spec.yaml config_rules.yaml image.tar instructions.md
+synapse.s9pk: manifest.yaml config_spec.yaml config_rules.yaml image.tar
 	appmgr -vv pack $(shell pwd) -o synapse.s9pk
```
Comment (elvece) [inline Makefile:12]: update to `embassy-sdk pack` and `embasy-sdk verify <app-id>.s9pk` - see [here](https://github.com/Start9Labs/cups-wrapper/pull/1/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R22) for an example

File: manifest.yaml [hunk 5]
```diff
@@ -1,38 +1,99 @@
...
+    entrypoint: config
+    args:
+      - "set"
+      - "/root"
+    mounts: {}
+    io-format: yaml
+dependencies: {}
+volumes: {}
+interfaces:
+  main:
```
Comment (elvece) [inline manifest.yaml:54]: add key:  `name` `description` to each interface (just "main" in this case)

File: Dockerfile [hunk 6]
```diff
@@ -27,6 +27,8 @@ ADD ./configurator.py /configurator.py
...
+RUN chmod a+x /usr/local/bin/config
```
Comment (elvece) [inline Dockerfile:31]: these two steps can be deleted in favor of using the `config` command in the `compat` image

File: Makefile [hunk 7]
```diff
@@ -8,16 +8,16 @@ all: synapse.s9pk
 install: synapse.s9pk
 	appmgr install synapse.s9pk
 
-synapse.s9pk: manifest.yaml config_spec.yaml config_rules.yaml image.tar instructions.md
+synapse.s9pk: manifest.yaml config_spec.yaml config_rules.yaml image.tar
```
Comment (elvece) [inline Makefile:11]: update to `embassy-sdk pack` and `embasy-sdk verify <app-id>.s9pk` - see [here](https://github.com/Start9Labs/cups-wrapper/pull/1/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R22) for an example

File: manifest.yaml [hunk 8]
```diff
@@ -1,38 +1,142 @@
...
+main:
+  type: docker
+  image: main
+  entrypoint: docker_entrypoint.sh
+  args: []
+  mounts:
+    main: /data
+    compat: /mnt/assets
+  io-format: yaml
+health-checks: {}
```
Comment (elvece) [inline manifest.yaml:29]: add health checks

File: config.sh (hunk 2 above)
Comment (elvece) [reply config.sh:1]: this still stands

File: Dockerfile (hunk 6 above)
Comment (elvece) [reply Dockerfile:31]: this still stands

File: manifest.yaml [hunk 9]
```diff
@@ -1,38 +1,142 @@
...
+      - /mnt/backup
+      - /data
+    mounts:
+      backup: /mnt/backup
+      main: /data
+  restore:
+    type: docker
+    image: compat
+    system: true
+    entrypoint: /mnt/assets/backup-restore.sh
```
Comment (elvece) [inline manifest.yaml:121]: does this script exist?

File: config.sh (hunk 2 above)
Comment (chrisguida) [reply config.sh:1]: done

File: Dockerfile (hunk 6 above)
Comment (chrisguida) [reply Dockerfile:31]: done

File: Makefile (hunk 7 above)
Comment (chrisguida) [reply Makefile]: done previously

File: manifest.yaml (hunk 8 above)
Comment (chrisguida) [reply manifest.yaml:29]: Need to get it running first

File: manifest.yaml (hunk 9 above)
Comment (chrisguida) [reply manifest.yaml:121]: No

**Review by kn0wmad — COMMENTED** [review]
Eyes on for science

File: Makefile [hunk 10]
```diff
@@ -1,23 +1,18 @@
...
-install: synapse.s9pk
-	appmgr install synapse.s9pk
+install: 
+	embassy-cli install synapse
 
-synapse.s9pk: manifest.yaml config_spec.yaml config_rules.yaml image.tar instructions.md
-	appmgr -vv pack $(shell pwd) -o synapse.s9pk
-	appmgr -vv verify synapse.s9pk
+verify: synapse.s9pk
+	embassy-sdk verify synapse.s9pk
```
Comment (elvece) [inline Makefile:12]:
```suggestion
	embassy-sdk verify s9pk synapse.s9pk
```

**Pull request merged**